presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriter.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.orc.OrcWriter;
+import com.facebook.presto.orc.metadata.CompressionKind;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.block.RunLengthEncodedBlock;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.OutputStreamSliceOutput;
+import io.airlift.slice.SliceOutput;
+import org.joda.time.DateTimeZone;
+
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_WRITER_CLOSE_ERROR;
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_WRITER_DATA_ERROR;
+import static com.facebook.presto.orc.OrcWriter.createOrcWriter;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class OrcFileWriter
+        implements HiveFileWriter
+{
+    private final OrcWriter orcWriter;
+    private final int[] fileInputColumnIndexes;
+    private final List<Block> nullBlocks;
+
+    public OrcFileWriter(
+            OutputStream outputStream,
+            List<String> columnNames,
+            List<Type> fileColumnTypes,
+            CompressionKind compression,
+            int[] fileInputColumnIndexes,
+            Map<String, String> metadata,
+            DateTimeZone hiveStorageTimeZone)
+    {
+        orcWriter = createOrcWriter(
```

Comment:
This file is exactly the same as `DwrfFileWriter` except this line. The other one says `createOrcWriter`. Consider merging them.